enhancements/installer/telemeter-metrics.md

```diff
@@ -0,0 +1,300 @@
+---
+title: installer-telemetry-metrics
+authors:
+  - "@patrickdillon"
+  - "@rna-afk"
+reviewers:
+  - "@abhinavdahiya"
+  - "@brancz"
+  - "@crawford"
+  - "@sdodson"
+approvers:
+  - "@abhinavdahiya"
+  - "@brancz"
+  - "@crawford"
+  - "@sdodson"
+creation-date: 2020-02-07
+last-updated: 2020-02-07
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# Installer Telemetry Metrics
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions
+
+ > 1. This requires creating an [aggregation pushgateway][weaveworks-aggregation-pushgateway] where the installer would push metrics. How would this be handled and by which team?
+ > 2. If #1 is possible, how would client authentication be administered to prevent [malicious pushing of metrics](risks-and-mitigations)? Authentication based on what? A pull secret? 
+ > 3. How to filter to collect only real user usage and prevent OpenShift developer/Red Hat internal users from skewing metrics? Perhaps filter by version?
```

Comment:
In any case we need to heavily confine the length of versions to produce very low cardinality something like `4.1.x`, `4.2.x`, `ci` etc. as all of this is going to be pushed to the push aggregation gateway as proposed which in turn will need to expose these on a metrics endpoint so we're talking an absolute upper bound within the 10 000s of time-series that we will ever be able to create for this, which we can reach very quickly.